src/components/MultiHopTrade/hooks/useHopHelper.tsx

```diff
@@ -0,0 +1,134 @@
+import { fromBaseUnit } from 'lib/math'
+import { SwapperName } from 'lib/swapper/api'
+import { selectFeeAssetById } from 'state/slices/assetsSlice/selectors'
+import {
+  selectPortfolioCryptoBalanceBaseUnitByFilter,
+  selectPortfolioCryptoPrecisionBalanceByFilter,
+} from 'state/slices/common-selectors'
+import { useAppSelector } from 'state/store'
+
+export const useHopHelper = () => {
+  const { selectedQuote } = useGetTradeQuotes()
+  const quoteData = useMemo(
+    () => (selectedQuote?.data?.isOk() ? selectedQuote.data.unwrap() : undefined),
+    [selectedQuote?.data],
+  )
+
+  const firstHop = useMemo(() => quoteData?.steps[0], [quoteData?.steps])
+  const lastHop = useMemo(() => quoteData?.steps[quoteData.steps.length - 1], [quoteData?.steps])
+
+  const firstHopSellAsset = useMemo(() => firstHop?.sellAsset, [firstHop?.sellAsset])
+  const firstHopBuyAsset = useMemo(() => firstHop?.buyAsset, [firstHop?.buyAsset])
+  const lastHopSellAsset = useMemo(() => lastHop?.sellAsset, [lastHop?.sellAsset])
+  const lastHopBuyAsset = useMemo(() => lastHop?.buyAsset, [lastHop?.buyAsset])
+
+  const { sellAssetAccountId: firstHopSellAssetAccountId } = useAccountIds({
+    buyAsset: firstHopBuyAsset,
+    sellAsset: firstHopSellAsset,
+  })
+
+  const { sellAssetAccountId: lastHopSellAssetAccountId } = useAccountIds({
+    buyAsset: lastHopBuyAsset,
+    sellAsset: lastHopSellAsset,
+  })
+
+  const firstHopSellFeeAsset = useAppSelector(state =>
+    selectFeeAssetById(state, firstHopSellAsset?.assetId ?? ''),
+  )
+
+  const lastHopSellFeeAsset = useAppSelector(state =>
+    selectFeeAssetById(state, lastHopSellAsset?.assetId ?? ''),
+  )
+
+  const firstHopFeeAssetBalanceFilter = useMemo(
+    () => ({ assetId: firstHopSellFeeAsset?.assetId, accountId: firstHopSellAssetAccountId ?? '' }),
+    [firstHopSellAssetAccountId, firstHopSellFeeAsset?.assetId],
+  )
+
+  const lastHopFeeAssetBalanceFilter = useMemo(
+    () => ({ assetId: lastHopSellFeeAsset?.assetId, accountId: lastHopSellAssetAccountId ?? '' }),
+    [lastHopSellAssetAccountId, lastHopSellFeeAsset?.assetId],
+  )
+
+  const firstHopFeeAssetBalancePrecision = useAppSelector(s =>
+    selectPortfolioCryptoPrecisionBalanceByFilter(s, firstHopFeeAssetBalanceFilter),
+  )
+
+  const lastHopFeeAssetBalancePrecision = useAppSelector(s =>
+    selectPortfolioCryptoPrecisionBalanceByFilter(s, lastHopFeeAssetBalanceFilter),
+  )
+
+  const sellAssetBalanceFilter = useMemo(
+    () => ({ accountId: firstHopSellAssetAccountId, assetId: firstHopSellAsset?.assetId ?? '' }),
+    [firstHopSellAssetAccountId, firstHopSellAsset?.assetId],
+  )
+
+  const sellAssetBalanceCryptoBaseUnit = useAppSelector(state =>
+    selectPortfolioCryptoBalanceBaseUnitByFilter(state, sellAssetBalanceFilter),
+  )
+
+  const sellAmountCryptoBaseUnit = useMemo(
+    () => firstHop?.sellAmountBeforeFeesCryptoBaseUnit,
+    [firstHop?.sellAmountBeforeFeesCryptoBaseUnit],
+  )
```

Comment:
not sure what our current policy is on use of useMemo (is it a blanket "always use it" or "use it when beneficial"?), but wrapping since property accesses like this reduces performance

Reply:
if this was in a selector it's as bad performance-wise but its acceptable since we're forced to use the same pattern for everything in redux